src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/RemoveUnusedLocals.Fixer.cs

```diff
@@ -1,27 +1,108 @@
 // Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System;
+using System.Composition;
+using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis.FindSymbols;
+using System.Collections.Generic;
+using System.Collections.Immutable;
 
 namespace Microsoft.CodeQuality.Analyzers.Maintainability
 {
     /// <summary>
     /// CA1804: Remove unused locals
     /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic, Name = AvoidUnusedPrivateFieldsAnalyzer.RuleId), Shared]
     public abstract class RemoveUnusedLocalsFixer : CodeFixProvider
     {
+        internal const string RuleId = "CA1804";
+
         public sealed override FixAllProvider GetFixAllProvider()
         {
             // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
             return WellKnownFixAllProviders.BatchFixer;
```

Comment:
I think we cannot use batch fixer. More specifically, consider the below case:

```
int a = 0, b = 0;
```

Both a and b are unused and hence we will get 2 unused local diagnostics. If user were to apply individual code fixes then the fix will only remove the declarators, either `a = 0` or `b = 0`, but a FixAll should remove the entire declaration statement. I think we need to implement our own custom FixAllProvider. I will sync with you offline.